Release 0.0.2

aburrell opened the pull request:
# Description

Release candidate for version 0.0.2.  This release also acts as a test for Issue #29.

# Type of change

- New release

# How Has This Been Tested?

This has been tested by running locally.

## Test Configuration
* Operating system: OS X Mojave
* Version number: Python 3.7
* Any details about your local setup that are relevant: using pysat develop-3 branch.

# Checklist:

- [x] Update the version number
- [x] Verify and update Zenodo information in .zeondo.json
- [x] doc tests pass locally
- [x] Upload the test distribution to the testPyPi server
- [x] Contacted developers and users about the future release
- [N/A] Performed integration tests with downstream packages (include list of these packages)
- [x] Create a wiki page with the release text (https://github.com/pysat/pysatSpaceWeather/wiki/Release-Draft-0.0.2)
- [ ] After approval, confirm all changes from the RC pull are listed in the wiki release text.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

jklenzing commented on .zenodo.json:
```diff
@@ -0,0 +1,27 @@
+{
+    "creators": [
+      {
+        "affiliation": "Goddard Space Flight Center",
+        "name": "Jeff Klenzing",
+        "orcid": "0000-0001-8321-6074"
+      }, 
+      {
+        "affiliation": "U.S. Naval Research Laboratory", 
+        "name": "Angeline G. Burrell",
+        "orcid": "0000-0001-8875-9326"
+      }, 
```
Does the order matter here?  Should we swap these moving forward?

aburrell replied:
Order does matter here, I suppose we should swap.

jklenzing reviewed (commented):
A couple of comments.  Working on the upload to the test pypi server.

aburrell commented on .zenodo.json:
```diff
@@ -0,0 +1,27 @@
+{
+    "creators": [
+      {
+        "affiliation": "Goddard Space Flight Center",
+        "name": "Jeff Klenzing",
+        "orcid": "0000-0001-8321-6074"
+      }, 
+      {
+        "affiliation": "U.S. Naval Research Laboratory", 
+        "name": "Angeline G. Burrell",
+        "orcid": "0000-0001-8875-9326"
+      }, 
```
```suggestion
    "creators": [
      {
        "affiliation": "U.S. Naval Research Laboratory", 
        "name": "Angeline G. Burrell",
        "orcid": "0000-0001-8875-9326"
      }, 
      {
        "affiliation": "Goddard Space Flight Center",
        "name": "Jeff Klenzing",
        "orcid": "0000-0001-8321-6074"
      }, 
```

jklenzing commented on docs/conf.py:
```diff
@@ -0,0 +1,189 @@
+#  author, documentclass [howto, manual, or own class]).
+latex_documents = [
+    (master_doc, 'pysatSpaceWeather.tex', u'pysatSpaceWeather Documentation',
+     u'Angeline G Burrell, Russel Stoneback, and Jeffery Klenzing', 'manual'),
```
Just realized we both need an update

jklenzing commented on docs/conf.py:
```diff
@@ -0,0 +1,189 @@
+#  author, documentclass [howto, manual, or own class]).
+latex_documents = [
+    (master_doc, 'pysatSpaceWeather.tex', u'pysatSpaceWeather Documentation',
+     u'Angeline G Burrell, Russel Stoneback, and Jeffrey Klenzing', 'manual'),
```
```suggestion
     u'Angeline G Burrell, Russell Stoneback, and Jeffrey Klenzing', 'manual'),
```

rstoneback commented:
Thanks @aburrell and @jklening. Meetings have resumed in force but I'm working on getting back to pysat and pysat* later this afternoon.

jklenzing commented:
Note that this will fail now due to dropped support of plot-related metadata.  See pysat/pysat#634

aburrell commented:
We forgot to do the downstream PRs 🤦

jklenzing commented:
I'll scan through for downstream PRs now.  The one for here is up.

aburrell commented:
Ok @rstoneback, would appreciate your approval/comments by Monday/merge-day.

jklenzing commented:
@aburell, my updates are done here.  ready to go unless @rstoneback has comments

rstoneback commented:
I'm having some problems with the pypi version that could be unique to my system somehow.
```
(standard3) russellstoneback@Russells-MacBook-Pro ~ % pip install -i https://test.pypi.org/simple/ pysatSpaceWeather
Looking in indexes: https://test.pypi.org/simple/
Collecting pysatSpaceWeather
  Downloading https://test-files.pythonhosted.org/packages/57/fb/5e6b5ba5e066f78d7067ad4e60bde346373fbbc4bbb9c436213d8bd6d2c0/pysatSpaceWeather-0.0.2-py3-none-any.whl (3.6 kB)
Installing collected packages: pysatSpaceWeather
Successfully installed pysatSpaceWeather-0.0.2
(standard3) russellstoneback@Russells-MacBook-Pro ~ % ipython
Python 3.8.3 (default, May 19 2020, 13:54:14) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.16.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import pysat                                                                                                                                                                                                        

In [2]: import pysatSpaceWeather                                                                                                                                                                                            
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-2-a948cfbe77a1> in <module>
----> 1 import pysatSpaceWeather

ModuleNotFoundError: No module named 'pysatSpaceWeather'
```
I can't get it to import from pip installation. pysatSpaceWeather shows up in `pip list` but not in `help('modules')` in python.
If I download the code manually and install it that way it works well. 

```
In [1]: import pysat                                                                                                                                                                                                        

In [2]: import pysatSpaceWeather                                                                                                                                                                                            

In [3]: pysat.utils.registry.register_by_module(pysatSpaceWeather.instruments)                                                                                                                                              

In [4]: pysat.user_modules                                                                                                                                                                                                  
Out[4]: 
{'sw': {'dst': 'pysatSpaceWeather.instruments.sw_dst',
  'f107': 'pysatSpaceWeather.instruments.sw_f107',
  'kp': 'pysatSpaceWeather.instruments.sw_kp',
  'ace': 'pysatSpaceWeather.instruments.sw_ace'}}                                                                                                                                                                             

In [6]: f107 = pysat.Instrument('sw', 'f107')                                                                                                                                                                               
```
I'm running on my working pull in pysat so the feedback from registry is improved.

rstoneback commented:
I tried to check out the docs on RTD but I didn't see an active build for this branch. You may want to turn on a build to ensure docs are working as expected there as well.

rstoneback commented on README.md:
````diff
@@ -30,17 +62,21 @@ python setup.py install
+Afterward, pysat will identify them using the `platform` and `name` keywords.
 
 ```
 import pysat
````
`import pysatSpaceWeather`

aburrell replied:
Actually, following the example above we don't need any more imports :P

aburrell replied:
Oh, I see.  Registry needs the module imported.

aburrell commented:
> I tried to check out the docs on RTD but I didn't see an active build for this branch. You may want to turn on a build to ensure docs are working as expected there as well.

Done!

aburrell commented:
> I'm having some problems with the pypi version that could be unique to my system somehow.

Did you check the path to see that 
- [ ] pypi did indeed install it
- [ ] the path is on you PYTHONPATH?

Those are all the ideas I have.

rstoneback commented on docs/supported_instruments.rst:
```diff
@@ -0,0 +1,54 @@
+Dst
+---
+
+The Disturbance Storm Time (Dst) Index is aa measure of magnetic activity
```
```suggestion
The Disturbance Storm Time (Dst) Index is a measure of magnetic activity
```

rstoneback commented on docs/supported_instruments.rst:
```diff
@@ -0,0 +1,54 @@
+
+Kp is a geomagnetic index thaat reflects the magnitude of geomagnetic
+disturbances at Earth.  Historic, recent (last 30 days), and forecasted values
+are available from the German Research Centrefor Geosciences at Potsdam,
```
```suggestion
are available from the German Research Centre for Geosciences at Potsdam,
```

rstoneback commented on docs/supported_instruments.rst:
```diff
@@ -0,0 +1,54 @@
+Kp
+---
+
+Kp is a geomagnetic index thaat reflects the magnitude of geomagnetic
```
```suggestion
Kp is a geomagnetic index that reflects the magnitude of geomagnetic
```

rstoneback commented on pysatSpaceWeather/instruments/methods/dst.py:
```diff
@@ -0,0 +1,41 @@
+    name : string
+        Name of space weather index, eg, dst, f107, kp
+    tag : string
+        Tag of the space waether index
```
```suggestion
        Tag of the space weather index
```

rstoneback commented on pysatSpaceWeather/instruments/methods/dst.py:
```diff
@@ -0,0 +1,41 @@
+    name : string
+        Name of space weather index, eg, dst, f107, kp
+    tag : string
+        Tag of the space waether index
```
```suggestion
        Tag of the space weather index
```

rstoneback reviewed (commented):
Gotta run off to a meeting. I'll be back though. Looking good. Found couple minor typos.

aburrell commented on README.md:
````diff
@@ -30,17 +62,21 @@ python setup.py install
+Afterward, pysat will identify them using the `platform` and `name` keywords.
 
 ```
 import pysat
````
```suggestion
```

aburrell commented on README.md:
````diff
@@ -30,17 +62,21 @@ python setup.py install
+instrument.
 
 ```
 import pysat
````
```suggestion
import pysat
import pysatSpaceWeather
```

rstoneback commented:
> > I'm having some problems with the pypi version that could be unique to my system somehow.
> 
> Did you check the path to see that
> 
> * [ ]  pypi did indeed install it
> * [ ]  the path is on you PYTHONPATH?
> 
> Those are all the ideas I have.

Thanks! I reinstalled pip via conda and now the pip install is good. Problem was on my end somewhere not on the package side.

rstoneback commented on README.md:
```diff
@@ -1,10 +1,38 @@
 
 # Installation
 
+The following instructions will allow you to install pysatSpaceWeather and give
```
```suggestion
The following instructions enable (?) you to install pysatSpaceWeather and give
```

aburrell replied:
Why the `(?)`?

rstoneback replied:
Trying to indicate it was a style suggestion though it didn't work.

rstoneback replied:
Longer explanation. First I noticed the voice was passive, and this is a nit here, plus the instructions don't actually allow folks to install the software, the license does. The first replacement word I came up with was enable, more active, but still suffered from not being fully correct. 
How about 'guide' instead?

aburrell replied:
Yeah, you have to put the question mark outside of the code suggestion box :P. Ok, I think your nitpick is very small and unimportant... BUT you are technically correct.  And if we know anything, it's that technical correctness is the best kind of correctness.

https://i.redd.it/8sdp12ekctz31.jpg

rstoneback commented on docs/supported_instruments.rst:
```diff
@@ -0,0 +1,54 @@
+F \ :sub:`10.7`\
+----------------
+
+F \ :sub:`10.7`\ is the 10.7 cm radio solar flux (measured in solar flux units,
```
```suggestion
F \ :sub:`10.7`\ is the 10.7 cm radio solar flux (measured in solar flux units,
```
The space after Fsub10.7 doesn't show up in the RTD version.

rstoneback commented on docs/methods.rst:
```diff
@@ -0,0 +1,47 @@
+information.
+
+
+.. automodule:: pysatSpaceWeather.instruments.methods.ace
```
Methods show up on RTD for Dst but not for Ace, F10.7, nor the kp ap pair..

aburrell replied:
It works flawlessly locally on this branch.

aburrell replied:
It may be an issue with the pysat dependence, since it's grabbing pysat from pip.

rstoneback commented on docs/examples/ex_init.rst:
```diff
@@ -0,0 +1,28 @@
+different historical, real-time, and forecasted solar, geomagnetic, and other
+space weather indices.  As specified in the
+`pysat tutorial <https://pysat.readthedocs.io/en/latest/tutorial.html>`_,
+data may be loaded using the following commands.  Historic F\ :sub:`10.7`\ is
```
Same space issue after f10.7 here as well.

rstoneback commented on docs/examples/ex_init.rst:
```diff
@@ -0,0 +1,28 @@
+   import pysatSpaceWeather as py_sw
+
+   old_time = dt.datetime(2012, 5, 14)
+   f107 = pysat.Instrument(inst_module=py_sw.instruments.sw_f107, tag='all',
+                           update_files=True)
```
This code goes a little outside the code box on RTD.

```suggestion
   f107 = pysat.Instrument(inst_module=py_sw.instruments.sw_f107, 
                           tag='all', update_files=True)
```

aburrell replied:
I think that's a function of your browser config, but I made the change.

rstoneback commented on docs/examples/ex_init.rst:
```diff
@@ -0,0 +1,28 @@
+   f107 = pysat.Instrument(inst_module=py_sw.instruments.sw_f107, tag='all',
+                           update_files=True)
+   f107.download(start=old_time)
+   f107.load(start=old_time)
```
```suggestion
   f107.load(date=old_time)
```

rstoneback commented on docs/examples/ex_init.rst:
```diff
@@ -0,0 +1,28 @@
+
+The output includes all available historic data (as implied by the tag name),
+including the specified date.  This data set starts on 14 February 1947 and
+will not reach up to the present day.
```
I think it would help users out if you included what the expected output should be here. I know you describe it, but the actual bits.

```
In [6]: print(f107.data)                                                                                                                                                                                                    
             f107
1947-02-14  253.9
1947-02-17  228.5
1947-02-19  178.8
1947-02-20  163.7
1947-02-24  164.1
...           ...
2018-04-27   69.6
2018-04-28   71.1
2018-04-29   72.2
2018-04-30   71.3

[25367 rows x 1 columns]
```

rstoneback commented on docs/history.rst:
```diff
@@ -0,0 +1,2 @@
+
+.. mdinclude:: ../CHANGELOG.md
```
I like the changelog in there.

rstoneback commented:
Something I noticed but may not need to be solved here.
The line below takes some time. Trying to find out how much.
```
In [7]: f107.load()            
```

rstoneback commented:
```
In [1]: import pysat                                                                                                                                                                                                        
im
In [2]: import pysatSpaceWeather                                                                                                                                                                                            

In [3]: f107 = pysat.Instrument('sw', 'f107')                                                                                                                                                                               

In [4]: f107.load()                                                                                                                                                                                                         

In [5]: f107.data                                                                                                                                                                                                           
Out[5]: 
             f107
2009-01-01   66.6
2009-01-02   67.6
2009-01-03   67.2
2009-01-04   66.6
...           ...
2015-08-27  112.5
2015-08-28  111.3
2015-08-29  102.1
2015-08-30   93.4
2015-08-31   92.7

[332 rows x 1 columns]

In [6]: f107 = pysat.Instrument('sw', 'f107', 'all')                                                                                                                                                                        

In [7]: f107.load()                                                                                                                                                                                                         

```
So the load on the first f107 returns quickly.
I'm currently 2 minutes and 3 GB into the blank load with tag='all'. The new pysat feature of loading everything at once is not playing nice with the existing features under that tag. 5 minutes in and 7 GB of memory.

aburrell commented:
Alright @rstoneback I think that addresses the issues you raised that are not related to pysat.  May want to put the F107 loading as an issue in pysat.
